### DISCUSSION TITLE
Question: is it possible to disable snapping for the layer that created by the same drawing tool?

### DISCUSSION THREAD

**Discussion by CityBelfast** [geoman-io/leaflet-geoman#852] [Questions & Help]
is there any way I can enable/disable the snapping for a specific layers. By default, I need to set enable Snapple option for the drawing tool.   But, I don't need the snap option for the layer that I've created by the tool. I've trying it in the  pm:create event. but it doesn't work. is there any way I could achieve it?

```
map.pm.setGlobalOptions({ snappable: true, snapDistance: 20, snapMiddle: true, allowSelfIntersection: false})
map.pm.disableDraw();
map.pm.addControls({ editMode: true, oneBlock: true, cutPolygon: false, drawCircleMarker: false });   

 map.on('pm:create', function (e) {
            var type = e.shape,
                layer = e.layer;
            layer.pm.disable({ allowSelfIntersection: false, snappable: false });
        });
```
https://user-images.githubusercontent.com/76213912/104387508-2e16c480-552f-11eb-843e-21b39084a427.MOV

**Comment by Falke-Design** [comment]
If I understand you correct: While drawing it should snap and while editing a layer it should not snap, then you can use this:

```
map.pm.addControls({ editMode: true, oneBlock: true, cutPolygon: false, drawCircleMarker: false }); 
map.pm.globalOptions.snappable = false;
```

**Comment by CityBelfast** [comment]
Let  me make it a little bit more clear about my question. Let us assume, I've a collection of features , which needs to be loaded along with the map. What I'm looking for here that I want snap those geometries only, which is loaded along with the map; I don't want to snap any other geometries that I've drawn manually on map by the drawing tool.

**Comment by Falke-Design** [comment]
Simply add `snapIgnore: true` to the new drawn layers:

```
 map.on('pm:create', function (e) {
            var type = e.shape,
                layer = e.layer;
            layer.options.snapIgnore = true
        });
```

**Comment by CityBelfast** [comment]
Thanks
